### PR TITLE
fix meshgrid binning

### PR DESCRIPTION
**Pull request by hoetzsch**
Meshgrid binning fix (by Zihao). To be nicified

### CODE REVIEW

File: appletree/likelihood.py [hunk 1]
```diff
@@ -52,8 +53,14 @@ def __init__(self, name: str = None, **config):
             warning = f'The usage of meshgrid binning is highly discouraged.'
             warn(warning)
             self.component_bins_type = 'meshgrid'
-            x_bins = jnp.linspace(*config['x_clip'], self._bins[0] + 1)
-            y_bins = jnp.linspace(*config['y_clip'], self._bins[1] + 1)
+            if type(self._bins[0]) == int:
```
[the remark this answers is not in the document]
Comment (dachengx) [reply appletree/likelihood.py:56]: Please use `isinstance()` here.

File: appletree/likelihood.py [hunk 2]
```diff
@@ -52,8 +53,14 @@ def __init__(self, name: str = None, **config):
...
+                x_bins = jnp.linspace(*config['x_clip'], self._bins[0] + 1)
+            else:
+                x_bins = jnp.array(self._bins[0])
+            if type(self._bins[1]) == int:
```
[the remark this answers is not in the document]
Comment (dachengx) [reply appletree/likelihood.py:60]: Also `isinstance()` here.

**Review by dachengx — CHANGES_REQUESTED** [review]
Thanks, @hoetzsch . After the requested changes are met, I think it is good to merge.

**Comment by hoetzsch** [comment]
@dachengx thanks, changed it!

**Review by dachengx — APPROVED** [review]
I added some tests and warnings. Looks good now.

**Pull request merged**